modules/core/src/shaderlib/shadow/shadow.js

```diff
@@ -133,16 +140,30 @@ function getViewportCenterPosition({viewport, center}) {
 function getViewProjectionMatrices({viewport, shadowMatrices}) {
   const projectionMatrices = [];
   const pixelUnprojectionMatrix = viewport.pixelUnprojectionMatrix;
-  const corners = [
-    [0, 0], // top left ground
-    [viewport.width, 0], // top right ground
-    [0, viewport.height], // bottom left ground
-    [viewport.width, viewport.height], // bottom right ground
-    [0, 0, -1.0], // top left near
-    [viewport.width, 0, -1.0], // top right near
-    [0, viewport.height, -1.0], // bottom left near
-    [viewport.width, viewport.height, -1.0] // bottom right near
-  ].map(pixel => screenToCommonSpace(pixel, pixelUnprojectionMatrix));
+  let corners = [];
+  if (viewport.isGeospatial) {
```

Comment:
`farDepth = viewports.isGeospatial ? 0 : 1` then only define `corners` once

Reply:
here it is a little different, without the depth z by default our API assumes the target z to be 0, not depth z to be zero.

Reply:
You can set it to undefined in that case